### PR TITLE
Stop redundant_array from creating non-Scope lifetime views, Fix MatMul

### PR DESCRIPTION
**Pull request by orausch**

### CODE REVIEW

File: tests/library/gemm_test.py [hunk 1]
```diff
@@ -14,6 +14,22 @@
 N = dace.symbol('N')
 
 
+@pytest.mark.parametrize(('implementation', ),
+                         [('pure', ), ('MKL', ),
+                          pytest.param('cuBLAS', marks=pytest.mark.gpu)])
+def test_gemm_no_c(implementation):
+    Gemm.default_implementation = implementation
```
Comment (tbennun) [inline tests/library/gemm_test.py:21]: please keep the old implementation and set it back. pytest does not create new python scripts when running.
Also, not everyone has MKL so you need to either mark it as such or have a try/except block

**Review by tbennun — CHANGES_REQUESTED** [review]

**Review by tbennun — APPROVED** [review]
LGTM if passes

**Comment by orausch** [comment]
Waiting on #590

**Comment by orausch** [comment]
merged in #597

**Pull request closed without merging**